chemprop/cli/train.py

```diff
@@ -1005,3 +1012,57 @@ def main(args):
     logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, force=True)
     args = parser.parse_args()
     TrainSubcommand.func(args)
+
+
+def summarize(task_type, dataset):
+    if task_type == "regression":
+        y = np.array([datapoint.y[0] for datapoint in dataset])
+        y_mean = y.mean()
+        y_std = y.std()
+        y_median = np.median(y)
+        mean_dev_abs = np.abs(y - y_mean)
+        num_1_sigma = (mean_dev_abs < y_std).sum()
+        num_2_sigma = (mean_dev_abs < 2 * y_std).sum()
+        percent_1_sigma = num_1_sigma / len(y) * 100
+        percent_2_sigma = num_2_sigma / len(y) * 100
+        column_headers = ["Statistic", "Value"]
+        table_rows = [
+            ["Num. datapoints", f"{len(y)}"],
+            ["Mean", f"{str(np.round(y_mean, 2))}"],
+            ["Std. dev.", f"{str(np.round(y_std, 2))}"],
+            ["Median", f"{str(np.round(y_median, 2))}"],
+            ["Datapoints within 1 std. dev.", f"{np.round(percent_1_sigma, 2)}%"],
+            ["Datapoints within 2 std. dev.", f"{np.round(percent_2_sigma, 2)}%"],
+        ]
+        return tuple([column_headers, table_rows])
+    elif task_type in ["classification", "multiclass"]:
+        y = np.array([datapoint.y[0] for datapoint in dataset])
+        mask = np.isnan(y)
+        classes = sorted(np.unique(y[~mask]))
+        class_counts = [(y == k).sum() for k in classes]
+        class_percents = [count / len(y) * 100 for count in class_counts]
+        nan_count = (mask).sum()
+        nan_percent = nan_count / len(y) * 100
+        column_headers = ["Class", "Count", "Percent"]
+        table_rows = []
+        for i, class_no in enumerate(classes):
+            table_row = [f"{class_no}", f"{class_counts[i]}", f"{np.round(class_percents[i], 2)}%"]
+            table_rows.append(table_row)
+        table_rows.append(["Nan", f"{nan_count}", f"{np.round(nan_percent, 2)}%"])
+        table_rows.append(["Total", f"{len(y)}", f"{100.00}%"])
+        return column_headers, table_rows
+    else:
+        return None
+
+
+def build_table(column_headers: list, table_rows: list) -> str:
+    table = Table(title="Summary of Training Data ")
+    for col in column_headers:
+        table.add_column(col)
```

Comment:
sometime later we should extend this to multitask datasets, but right now assuming single-task is fine
```suggestion
    table = Table(*column_headers, title="Summary of Training Data (task 0)")
```